### ISSUE TITLE
Helm: Add Model Catalog registration for trained models

### ISSUE THREAD

**Issue by tosin2013** [KubeHeal/openshift-aiops-platform#128]
## Context

RHOAI 3.5 Model Catalog provides centralized model governance. The platform's models are only registered implicitly via KServe InferenceService CRs.

### New template: model-catalog-registration.yaml
Job or Tekton task that registers models in the RHOAI Model Catalog after successful training + evaluation:
- Model name, version, framework (scikit-learn, TensorFlow)
- Training metadata (date, dataset, accuracy metrics)
- Artifact location (S3 path or PVC)
- Link to deployed InferenceService

### tekton-model-training-pipeline.yaml
Add catalog registration step:
```
Train → Evaluate (EvalHub) → Register (Model Catalog) → Deploy (canary)
```

### values.yaml
```yaml
modelCatalog:
  enabled: false
  autoRegister: true
```

## Depends on
- #115 (Evaluate Model Catalog) — must confirm custom model registration
- #125 (EvalHub integration) — evaluation metrics feed into catalog metadata
- #114 (MLflow evaluation) — MLflow may overlap with catalog

## Acceptance criteria

- [ ] Models appear in RHOAI Model Catalog UI
- [ ] Version history tracked across training runs
- [ ] Registration automated in training pipeline
